source/mir/random/flex/internal/area.d

```diff
@@ -0,0 +1,668 @@
+module mir.random.flex.internal.area;
+
+import mir.random.flex.internal.types : Interval;
+import std.traits : ReturnType;
+
+/**
+Determines the hat and squeeze function of an interval.
+Based on Theorem 1 of Botts et al. (2013).
```

Comment:
This documentation should make clear that the hat and squeeze functions are set in-place in the `Interval` instance provided to the function.